src/student_success_tool/preprocessing/pdp.py

```diff
@@ -238,11 +238,29 @@ def add_empty_cols_if_missing(
     )
 
 
-def clean_up_labeled_dataset_cols_and_vals(df: pd.DataFrame) -> pd.DataFrame:
+def clean_up_labeled_dataset_cols_and_vals(
+    df: pd.DataFrame, num_credits_col: str = "num_credits_earned_cumsum"
+) -> pd.DataFrame:
     """
     Drop a bunch of columns not needed or wanted for modeling, and set to null
     any values corresponding to time after a student's current year of enrollment.
+
+    Args:
+        df: DataFrame as created with features and targets and limited to the checkpoint term.
+        num_credits_col: Name of the column containing cumulative earned credits.
     """
+    num_credit_check = constants.DEFAULT_COURSE_CREDIT_CHECK
+    credit_pattern = re.compile(rf"in_{num_credit_check}_credits")
+    # To prevent data leakage, students that have not reached the 12 credits and not taken the course
+    # by the checkpoint term (which this data is limited to at the time of this function),
```

Comment:
@kaylawilding why are we only setting to null if the column is False? Even if the value is True, it should still be null if the student hasn't yet reached 12 credits, b/c that's the data leakage then... am I misunderstanding?

Reply:
if you haven't yet reached 12 credits, but you took the course after the checkpoint - you would still have a true value, right? and that is info beyond the checkpoint which is leakage